### DISCUSSION TITLE
Failed to load resource: the server responded with a status of 404 (Not Found)

### DISCUSSION THREAD

**Discussion by bomber92** [vimeo/player.js#864] [Q&A]
Guest user get an 404 error when access to my VOD's main feature video.
Where should I check?
![20221023](https://user-images.githubusercontent.com/116472306/197370953-c175d4cd-801e-4543-8234-e34bea5b8e3c.jpg)

**Comment by fisherinnovation** [comment]
Can you provide links or examples of this issue to aid in triage and debugging?

**Comment by bomber92** [comment]
Thanks for your reply.

Current Error is
Failed to load resource: the server responded with a status of 500 (Internal Server Error)

test page is 
https://www.kiralash.jp/

**Comment by Tyru5** [comment]
Hey @bomber92!

Hope you are having a great day so far ☀️! There are a few things to take into account here, and below you'll find my observations after looking into your issue further:

Vimeo provides a wide range of video privacy settings, ranging from public, unlisted, private, etc... _**as well as**_ embeddable specific options. These can be found in the video settings slideout, when you click on a video.

I believe the error you are seeing on your page when it loads, comes down to an embeddable option set on the video itself -- more specifically, I believe you have the embeddable option set to `specific domains` and you have not provided the necessary sites to whitelist so they Vimeo video could be embedded.

Here is a visual of what I'm referencing:
<img width="624" alt="Screen Shot 2022-10-26 at 11 59 07 AM" src="https://user-images.githubusercontent.com/11080409/198102581-07e6c18a-ba08-4371-b240-115f4b423981.png">

Additionally -- reference the `oembed` data that is being sent back for this specific video of yours:

URL: `https://vimeo.com/api/oembed.json?url=https://vimeo.com/764002059?h=5f367f80a2`
```
{
   "type": "video",
   "version": "1.0",
   "provider_name": "Vimeo",
   "provider_url": "https://vimeo.com/",
   "html": "<iframe src=\"https://player.vimeo.com/video/764002059?h=5f367f80a2&amp;app_id=122963\" width=\"426\" 
   height=\"240\" frameborder=\"0\" allow=\"autoplay; fullscreen; picture-in-picture\" allowfullscreen></iframe>",
   "width": 426,
   "height": 240,
   "domain_status_code": 403, <-- This jumps out at me, and aligns with the issue we are seeing
   "video_id": 764002059,
   "uri": "/videos/764002059"
}
```

It looks like to me, that the video you are trying to use has its' embed option set to `specific domain`, and you have not added your site(s) as a valid domain to embed the video on, hence the error.

The solution to this would be to add your site as a valid domain so that the video may be embedded:
<img width="479" alt="Screen Shot 2022-10-26 at 12 09 13 PM" src="https://user-images.githubusercontent.com/11080409/198103346-9e84bf50-e38a-4ba8-9afd-85b0b0df3be0.png">

If you are curious on how to work with oEmbed specifically with Vimeo, more information could be referenced here:
- https://developer.vimeo.com/api/oembed/videos

There is also a specific section detailing and explaining how to work with Vimeo videos that have domain-level privacy enabled on them which could be referenced here:
- https://developer.vimeo.com/api/oembed/videos#embedding-videos-with-domain-privacy

Hope this helps in some way!

Best,
Ti 🚀

**Comment by bomber92** [comment]
Thanks for your reply.

I tried but the video did not watched.
![20221027](https://user-images.githubusercontent.com/116472306/198164830-d1327cbb-23a0-4baf-8aeb-8992197eeca2.png)

**Comment by bomber92** [comment]
Different embedding case.
![20221027-02](https://user-images.githubusercontent.com/116472306/198167032-30e451b4-2f04-42de-aa91-c26f0a595051.png)
